internal/schemavalidators/schemavalidators.go

```diff
@@ -93,3 +126,39 @@ func ValidateCycloneDX(data interface{}, version CycloneDXVersion) error {
 
 	return nil
 }
+
+// ValidateCSAF validates the given object against a CSAF schema version.
+// The schema version is determined by the "csaf_version" field in the object.
+func ValidateCSAF(data interface{}) error {
+	var errs error
+	err := compiledCSAFSchemas[CSAFVersion2_1].Validate(data)
+	if err != nil {
+		if err := errorIsJSONFormat(err); err != nil {
+			return err
+		}
+
+		errs = multierror.Append(errs, err)
+	} else {
+		return nil
+	}
+
+	err = compiledCSAFSchemas[CSAFVersion2_0].Validate(data)
+	if err != nil {
+		if err := errorIsJSONFormat(err); err != nil {
```

Comment:
this might not be needed, as it's already checked above.

Reply:
The point is that if we don't add it with the multi error, the same error regarding the format will appear twice on this way, if that's the problem, it will just exist on the first try.